kernel/src/fs/epoll/epoll_file.rs

```diff
@@ -215,6 +221,47 @@ impl EpollFile {
         }
     }
 
+    fn collect_ready(&self, max_events: usize, ep_events: &mut Vec<EpollEvent>) {
```

Comment:
Why spilting the original `pop_ready` as the new `collect_ready` and `pop_ready`? I scan the code, the new `pop_ready` is only used in `collect_ready` now. 

The parameter `limit` of `pop_ready` makes me feel strange. If these two methods are merged in one, we do not need this parameter. The new `pop_ready` actually pops at most one ready entry now.

If you think spliting is better, the methods `collect_ready` and `pop_ready` may be better named as `pop_all_ready` and `pop_one_ready` to show their relationship.

Reply:
Well, I think there are some reasons to justify splitting them:

- Code length: `pop_ready` and `collect_ready` are both >40 lines. It will be about ~100 lines. This is a bit longer, and if it can be split, I would prefer to leave it split.
- Code complexity: As you can see, both `pop_ready` and `collect_ready` now have a loop. If we merge them, there will be nested loops. Again, if it can be avoided, I would rather avoid it.
- Consistency: Since `push_ready` pushes exactly _one_ ready event, it makes a lot of sense to have a `pop_ready` method that pops exactly _one_ ready event.
- Modularization: `push_ready`/`pop_ready` keeps the synchronization of the `is_ready` flag and whether the item in the list is ready or not. This seems better than having the logic mixed into another complicated logic.

> If you think spliting is better, the methods `collect_ready` and `pop_ready` may be better named as `pop_all_ready` and `pop_one_ready` to show their relationship.

`collect` may be confusing. Perhaps renaming `collect_ready` to `pop_multi_ready`? It's not "all" in fact, as it accepts a `max_events` parameter.

Reply:
You words make sense to me! I also feel spliting these two methods is better now.

> collect may be confusing. Perhaps renaming collect_ready to pop_multi_ready?

`pop_multi_ready` looks fine to me.